tkg/client/cluster.go

```diff
@@ -862,3 +874,63 @@ func (c *TkgClient) ValidateManagementClusterVersionWithCLI(regionalClusterClien
 
 	return nil
 }
+
+// validateConfigForSingleNodeCluster validates that the controlPlaneTaint CC variable is not set for the single node workload cluster, otherwise returns error
+func validateConfigForSingleNodeCluster(stream []byte, options *CreateClusterOptions, tkgClient *TkgClient) error {
+	cluster, err := getClusterObjectFromYaml(stream)
+	if err != nil {
+		return err
+	}
+
+	if !topology.IsSingleNodeCluster(cluster) {
+		return nil
+	}
+
+	controlPlaneTaint := true
+	err = topology.GetVariable(cluster, "controlPlaneTaint", &controlPlaneTaint)
+	if err != nil {
+		return errors.Wrap(err, "failed to get CC variable controlPlaneTaint")
+	}
+
+	// Do not allow the creation of single node clusters without the feature gate.
+	if !tkgClient.IsFeatureActivated(config.FeatureFlagSingleNodeClusters) {
+		return errors.New("Worker count cannot be 0, minimum worker count required is 1")
+	}
+
+	if controlPlaneTaint {
+		return errors.New(fmt.Sprintf("unable to create single node cluster %s as control plane node has taint", options.ClusterName))
+	}
+
+	return nil
+}
+
+func getClusterObjectFromYaml(stream []byte) (*capi.Cluster, error) {
```

Comment:
we don't have a method that does this already?

Reply:
Didnt find one, thats why used own helper.